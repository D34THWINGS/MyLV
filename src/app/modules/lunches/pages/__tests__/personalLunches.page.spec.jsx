import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { PersonalLunchesPage } from '../personalLunches.page'

jest.unmock('../personalLunches.page')

describe('PersonalLunchesPage', () => {
  let props

  const getWrapper = () => shallow(<PersonalLunchesPage {...props} />)

  beforeEach(() => {
    props = {
      lunches: [{
        id: 'foo',
      }],
      fetchUserLunches: jest.fn(),
      deleteLunch: jest.fn(),
      isLoading: false,
    }
  })

  it('should render loader if loading', () => {
    // Given
    props.isLoading = true

    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should render lunch list if not loading', () => {
    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
