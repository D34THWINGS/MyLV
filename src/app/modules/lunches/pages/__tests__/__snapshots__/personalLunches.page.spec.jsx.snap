// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalLunchesPage should render loader if loading 1`] = `<WithStyles(LoadingPage) />`;

exports[`PersonalLunchesPage should render lunch list if not loading 1`] = `
<Card
  raised={false}
>
  <WithStyles(CardContent)>
    <WithStyles(Typography)
      component="h2"
      gutterBottom={true}
      variant="headline"
    >
      Mes déjeuners
    </WithStyles(Typography)>
    <WithStyles(Typography)
      gutterBottom={true}
    >
      Merci de renseigner ici les déjeuners en équipes payés par LinkValue. Ceci a pour but d'éviter de verser des tickets restaurants pour des déjeuners déjà remboursés.
    </WithStyles(Typography)>
    <WithStyles(Table)>
      <TableHead
        component="thead"
      >
        <WithStyles(TableRow)>
          <WithStyles(TableCell)>
            Intitulé
          </WithStyles(TableCell)>
          <WithStyles(TableCell)>
            Date
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            numeric={true}
          >
            Participants
          </WithStyles(TableCell)>
        </WithStyles(TableRow)>
      </TableHead>
      <TableBody
        component="tbody"
      >
        <WithStyles(LunchRow)
          key="foo"
          lunch={
            Object {
              "id": "foo",
            }
          }
          onLunchDelete={[MockFunction]}
        />
      </TableBody>
    </WithStyles(Table)>
  </WithStyles(CardContent)>
  <WithStyles(CardActions)>
    <WithStyles(Button)
      color="primary"
      component={[Function]}
      size="small"
      to="/lunches/new"
    >
      Ajouter un déjeuner
    </WithStyles(Button)>
  </WithStyles(CardActions)>
</Card>
`;
